lib/matplotlib/backends/tkagg.py

```diff
@@ -13,23 +13,24 @@ def blit(photoimage, aggimage, bbox=None, colormode=1):
 
     if bbox is not None:
         bbox_array = bbox.__array__()
+        # x1, x2, y1, y2
+        bboxptr = (bbox_array[0, 0], bbox_array[1, 0], bbox_array[0, 1], bbox_array[1, 1])
     else:
-        bbox_array = None
+        bboxptr = 0
     data = np.asarray(aggimage)
+    dataptr = (data.ctypes.data, data.shape[0], data.shape[1])
     try:
         tk.call(
             "PyAggImagePhoto", photoimage,
-            id(data), colormode, id(bbox_array))
+            dataptr, colormode, bboxptr)
     except Tk.TclError:
-        try:
-            try:
-                _tkagg.tkinit(tk.interpaddr(), 1)
-            except AttributeError:
-                _tkagg.tkinit(id(tk), 0)
-            tk.call("PyAggImagePhoto", photoimage,
-                    id(data), colormode, id(bbox_array))
-        except (ImportError, AttributeError, Tk.TclError):
-            raise
+        if hasattr(tk, 'interpaddr'):
+            _tkagg.tkinit(tk.interpaddr(), 1)
+        else:
+            # very old python?
+            _tkagg.tkinit(tk, 0)
```

Comment:
interpaddr was added in 1998(! https://github.com/python/cpython/commit/9d1b7ae) so I cannot actually imagine a case where this is triggered today (but that can go to another PR of course)